staging/src/k8s.io/apiserver/pkg/storage/cacher/ready_test.go

```diff
@@ -38,16 +74,47 @@ func Test_newReady(t *testing.T) {
 	}
 }
 
+// Test_newReadyRacy executes all the possible transitions randomly.
+// It must run with the race detector enabled.
+func Test_newReadyRacy(t *testing.T) {
```

Comment:
this test is flaking (e.g. https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116222/pull-kubernetes-unit/1631324028675297280)

https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-unit&width=20&include-filter-by-regex=cacher

https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=apiserver%2Fpkg%2Fstorage%2Fcacher#ea6679417165f10786e6

Reply:
hitting ~3% failure rate with stress

```
go test -race -c k8s.io/apiserver/pkg/storage/cacher
stress ./cacher.test -test.run Test_newReadyRacy
```

Reply:
opened https://github.com/kubernetes/kubernetes/issues/116225 to track

Reply:
hmm, I did use stress , let me check

Reply:
is not a flake, there was one test that timeout out and panic

```
panic: test timed out after 3m0s
running tests:
	Test_newReadyRacy (2m8s)
```

at least I can't reproduce it
```
4m35s: 791173 runs so far, 0 failures
4m40s: 805293 runs so far, 0 failures
4m45s: 819501 runs so far, 0 failures
4m50s: 833768 runs so far, 0 failures
4m55s: 848068 runs so far, 0 failures
5m0s: 862304 runs so far, 0 failures
```

and these links doesn't work for me 🤔 

https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-unit&width=20&include-filter-by-regex=cacher

https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=apiserver%2Fpkg%2Fstorage%2Fcacher#ea6679417165f10786e6

Reply:
https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=apiserver%2Fpkg%2Fstorage%2Fcacher is a good link

looks like the testgrid history got truncated today

Reply:
ok, but this is a different one
```
Failed
=== RUN   TestGetCurrentResourceVersionFromStorage
panic: Object 'Kind' is missing in '{"name":"pod-1","namespace":"ns","creationTimestamp":null,"Spec":{"RestartPolicy":"","TerminationGracePeriodSeconds":null,"ActiveDeadlineSeconds":null,"NodeSelector":null,"ServiceAccountName":"","NodeName":"","Hostname":"","Subdomain":"","SchedulerName":""},"Status":{"Phase":"","Conditions":null,"Message":"","Reason":"","HostIP":"","PodIP":"","StartTime":null}}
'

```

Reply:
Fixed in https://github.com/kubernetes/kubernetes/issues/116292